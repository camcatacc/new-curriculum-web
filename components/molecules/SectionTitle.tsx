import React from "react";
import { SectionTitleText } from "../atoms/SectionTitleText";
import { SectionTitleDivider } from "../atoms/SectionTitleDivider";
import { InlineStyles } from "../../styles/stylesInterface";

export const SectionTitle = (props: { children: string }) => {
    return (
        <div className="flex flex-col mt-8 mb-8">
            <SectionTitleText>{props.children}</SectionTitleText>
            <SectionTitleDivider />
        </div>
    );
};
