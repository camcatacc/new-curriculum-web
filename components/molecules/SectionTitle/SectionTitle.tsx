// Modules
import React from "react";

// Elements
import SectionTitleDivider from "components/atoms/SectionTitleDivider/SectionTitleDivider";
import SectionTitleText from "components/atoms/SectionTitleText/SectionTitleText";

// Definitions
export interface SectionTitleProps {
	children: string;
	animated?: boolean;
}

// Components
const SectionTitle = ({ children, animated }: SectionTitleProps) => {
	return (
		<div className="flex flex-col mt-8 mb-8">
			<SectionTitleText animated={animated}>{children}</SectionTitleText>
			<SectionTitleDivider animated={animated} />
		</div>
	);
};

export default SectionTitle;
