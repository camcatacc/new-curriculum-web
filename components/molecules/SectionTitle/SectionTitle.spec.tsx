// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import SectionTitle from "components/molecules/SectionTitle/SectionTitle";

// Definitions
import type { SectionTitleProps } from "components/molecules/SectionTitle/SectionTitle";

// Default Props
const defaultProps: SectionTitleProps = {
	children: "This is a Section Title"
};

// Tests
describe("SectionTitle", () => {
	it("Renders as expected", () => {
		const { container } = render(<SectionTitle {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
