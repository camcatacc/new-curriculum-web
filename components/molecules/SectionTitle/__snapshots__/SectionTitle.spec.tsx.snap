// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle Renders as expected 1`] = `
<div>
  <div
    class="flex flex-col mt-8 mb-8"
  >
    <div>
      <h2
        class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
      >
        This is a Section Title
      </h2>
    </div>
    <div
      class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
    />
  </div>
</div>
`;
